dataactcore/utils/fileE.py

```diff
@@ -0,0 +1,101 @@
+from collections import namedtuple
+import logging
+from operator import attrgetter
+
+from suds.client import Client
+
+from dataactcore.config import CONFIG_BROKER
+
+
+logger = logging.getLogger(__name__)
+
+
+def configValid():
+    """Does the config have the necessary bits for talking to the SAM SOAP
+    API"""
+    sam = CONFIG_BROKER.get('sam') or {}
+    hasWsdl = bool(sam.get('wsdl'))
+    hasUser = bool(sam.get('username'))
+    hasPass = bool(sam.get('password'))
+    return hasWsdl and hasUser and hasPass
+
+
+def createAuth(client):
+    auth = client.factory.create('userAuthenticationKeyType')
+    auth.userID = CONFIG_BROKER['sam']['username']
+    auth.password = CONFIG_BROKER['sam']['password']
+    return auth
+
+
+def createSearch(client, duns):
+    search = client.factory.create('entitySearchCriteriaType')
+    search.DUNSList = client.factory.create('DUNSList')
+    search.DUNSList.DUNSNumber = duns
+    return search
+
+
+def getEntities(client, duns):
+    """Hit the SAM SOAP API, searching for the provided DUNS numbers. Return
+    the results as a list of Suds objects"""
+    params = client.factory.create('requestedData')
+    params.coreData.value = 'Y'
+
+    result = client.service.getEntities(
+        createAuth(client), createSearch(client, duns), params)
+
+    if result.listOfEntities:
+        return result.listOfEntities.entity
+    else:
+        return []
+
+
+Row = namedtuple('Row', (
+    'AwardeeOrRecipientUniqueIdentifier',
+    'UltimateParentUniqueIdentifier',
+    'UltimateParentLegalEntityName',
+    'HighCompOfficer1Name',
```

Comment:
That's been discussed out of band; the IDD's going to be changed as the web service doesn't break up names that way.